app/models/post.rb

```diff
@@ -1127,6 +1127,10 @@ def cannot_permanently_delete_reason(user)
     end
   end
 
+  def self.show_welcome_topic_banner?
+    Post.find_by("topic_id = :topic_id AND post_number = 1 AND version = 1 AND created_at > :created_at", topic_id: SiteSetting.welcome_topic_id, created_at: 1.month.ago).present?
```

Comment:
Are we able to use a hidden site setting to indicate whether the banner should be displayed instead? Otherwise, every `/latest` request going forward has to incur the overhead of this query.